chsdi/templates/luftbilder/viewer.mako

```diff
@@ -260,7 +279,58 @@
         view.on('propertychange', debounce(updateUrl));
         updateUrl();
 
+        tile.on('postcompose', function(event) {
+          filter(event.context, lubisMap);
+        });
+
+        function filter(context, lubisMap) {
+          var canvas = context.canvas;
+          var brightnessSlider = document.getElementById("brightness");
+          var contrastSlider = document.getElementById("contrast");
+          var saturationSlider = document.getElementById("saturation");
+          brightnessSlider.addEventListener("change", brightnessFunction(canvas, brightnessSlider.value, lubisMap));
+          contrastSlider.addEventListener("change", contrastFunction(canvas, contrastSlider.value, lubisMap));
+          saturationSlider.addEventListener("change", saturationFunction(canvas, saturationSlider.value, lubisMap));
+        }
       }
+
+      var brightnessFunction = function(canvas, brightnessVal, map) {
+        if (map){
+          var context = canvas.getContext('2d');
+          context.filter = "brightness(" + brightnessVal + "%)";
+          context.drawImage(canvas, 0, 0);
+          map.render();
+        }
+      };
+
+      var contrastFunction = function(canvas, contrastVal, map) {
+        if (map){
+          var context = canvas.getContext('2d');
+          context.filter = "contrast(" + contrastVal + "%)";
+          context.drawImage(canvas, 0, 0);
+          map.render();
+        }
+      };
+
+      var saturationFunction = function(canvas, saturationVal, map) {
+        if (map){
+          var context = canvas.getContext('2d');
+          context.filter = "saturate(" + saturationVal + "%)";
+          context.drawImage(canvas, 0, 0);
+          map.render();
+        }
+      };
```

Comment:
These 3 functions seems pretty similar , You can factorize them to only  use 1 function.